app/controllers/posts_controller.rb

```diff
@@ -6,4 +6,31 @@ def index
   def show
     @post = User.find(params[:user_id]).posts.find(params[:id])
   end
+
+  def create
+    new_post = current_user.posts.new(post_params)
+    new_post.comments_counter = 0
+    new_post.likes_counter = 0
+    @post = new_post
+
+    respond_to do |format|
+      if new_post.save
+        format.html do
+          flash[:success] = 'Post created successfully'
+          redirect_to user_posts_url
+        end
+      else
+        format.html do
+          flash.now[:error] = 'Error: Post could not be created'
+          render :new, locals: { post: new_post }
```

Comment:
Dear @FrederickMih ,
I would like to inform you that I have completed all the tasks outlined in the project requirements. With all due respect, based on my thorough assessment, I firmly believe that the comments you provided should be treated as suggested changes rather than mandatory modifications. I have taken into account the project specifications and executed the necessary actions accordingly.